packages/next/test/integration/integration-1.test.js

```diff
@@ -122,6 +122,40 @@ if (parseInt(process.versions.node.split('.')[0], 10) >= 16) {
     // );
   });
 
+  it('should build with app-dir with segment options correctly', async () => {
+    const { buildResult } = await runBuildLambda(
+      path.join(__dirname, '../fixtures/00-app-dir-segment-options')
+    );
+
+    const lambdas = new Set();
+
+    for (const key of Object.keys(buildResult.output)) {
+      if (buildResult.output[key].type === 'Lambda') {
+        lambdas.add(buildResult.output[key]);
+      }
+    }
+
+    expect(
+      buildResult.routes.some(
+        route =>
+          route.src?.includes('_next/data') && route.src?.includes('.rsc')
+      )
+    ).toBeFalsy();
+
+    expect(lambdas.size).toBe(2);
+
+    expect(buildResult.output['api/hello']).toBeDefined();
+    expect(buildResult.output['api/hello'].type).toBe('Lambda');
+    expect(buildResult.output['api/hello'].maxDuration).toBe(7);
+
+    expect(buildResult.output['api/hello-again']).toBeDefined();
+    expect(buildResult.output['api/hello-again'].type).toBe('Lambda');
+    expect(buildResult.output['api/hello-again'].maxDuration).toBe(7);
+    expect(
+      buildResult.output['api/hello-again'].supportsResponseStreaming
+    ).toBe(true);
```

Comment:
suggestion: Is it intentional not to add this assertion for `api/hello`?

Reply:
As far as I know only app dir supports streaming?

Reply:
I thought `api/hello` is also in app dir. Looks good as is! 👍